.github/workflows/publish-jsr.yml

```
@@ -48,19 +52,21 @@ jobs:
         run: |
           cmake --build build --config Release
 
-      - name: Embed .so as JSON/base64 for JSR
+      - name: Embed binary as JSON/base64 for JSR
         run: |
-          deno run --allow-read --allow-write jsr/scripts/embed_so.ts \
+          ./jsr/scripts/embed_binary.sh \
             build/libcpp_bindings_linux.so \
-            jsr/bin/x84_64.json \
             linux-x86_64
+          cp LICENSE ./jsr/package/LICENSE
 
-      - name: Publish package to JSR
-        working-directory: jsr
+      - name: Publish package to JSR (real)
+        if: github.ref_name == 'main' || startsWith(github.ref, 'refs/tags/')
```

why should this only be run on main?


```suggestion
        if: github.ref_type == 'tag'
```

---

Why would you publish a package that got tagged from a "not main" branch?

---

These packages should only be published from main

---

When you have a tag 2.0.0 and have to-do a patch for version 1.4.0 you will create a branch from 1.4.0 and set the tag on that branch, and it is not main then

---

Your version would mean I can only support the latest version, and apply patches there. Therefore forcing someone to take braking changes potentially